src/store/selectors/deck-filters.ts

```diff
@@ -0,0 +1,117 @@
+import { createSelector } from "reselect";
+
+import { assert } from "@/utils/assert";
+import { FACTION_ORDER } from "@/utils/constants";
+import { and, or } from "@/utils/fp";
+import uFuzzy from "@leeoniya/ufuzzy";
+import type { ResolvedDeck } from "../lib/types";
+import type { StoreState } from "../slices";
+import type { DeckFiltersType } from "../slices/deck-collection-filters.types";
+import type { MultiselectFilter } from "../slices/lists.types";
+import { selectLocalDecks } from "./decks";
+
+// Arbitrarily chosen for now
+const MATCHING_MAX_TOKEN_DISTANCE_DECKS = 4;
+
+const filterDeckByFaction = (faction: string) => {
+  return (deck: ResolvedDeck) => {
+    return deck.cards.investigator.card.faction_code === faction;
+  };
+};
+
+const makeDeckFactionFilter = (values: MultiselectFilter) => {
+  return or(values.map((value) => filterDeckByFaction(value)));
+};
+
+const makeDeckFilterFunc = (userFilters: DeckFiltersType) => {
+  const filterFuncs = [];
+
+  for (const filter of Object.keys(userFilters)) {
+    switch (filter) {
+      case "faction":
+        filterFuncs.push(makeDeckFactionFilter(userFilters[filter]));
+    }
+  }
+
+  return and(filterFuncs);
+};
+
+export const selectDeckFilters = createSelector(
```

Comment:
`createSelector` is only necessary for things that change referential quality. It is not necessary for selectors that return a primitive data type (string, number) or things that return a direct reference to something in the store.